build.go

```diff
@@ -90,107 +80,217 @@ func Build(projectPathParser PathParser,
+
+				layer.SBOM, err = sbomContent.InFormats(context.BuildpackInfo.SBOMFormats...)
+				if err != nil {
+					return packit.BuildResult{}, err
+				}
+			} else {
+				logger.Process("Reusing cached layer %s", layer.Path)
+				err := os.RemoveAll(filepath.Join(projectPath, "node_modules"))
+				if err != nil {
+					return packit.BuildResult{}, err
+				}
+
+				err = os.Symlink(filepath.Join(layer.Path, "node_modules"), filepath.Join(projectPath, "node_modules"))
+				if err != nil {
+					return packit.BuildResult{}, err
+				}
 			}
+			layer.Build = true
+			layer.Cache = true
 
-			err = environment.Configure(nodeModulesLayer, globalNpmrcPath)
-			if err != nil {
-				return packit.BuildResult{}, err
-			}
-
-			logger.EnvironmentVariables(nodeModulesLayer)
-
-			logger.GeneratingSBOM(nodeModulesLayer.Path)
+			layers = append(layers, layer)
+		}
 
-			var sbomContent sbom.SBOM
-			duration, err = clock.Measure(func() error {
-				sbomContent, err = sbomGenerator.Generate(context.WorkingDir)
-				return err
-			})
+		if launch {
+			layer, err := context.Layers.Get("launch-modules")
 			if err != nil {
 				return packit.BuildResult{}, err
 			}
-			logger.Action("Completed in %s", duration.Round(time.Millisecond))
-			logger.Break()
 
-			logger.FormattingSBOM(context.BuildpackInfo.SBOMFormats...)
-
-			nodeModulesLayer.SBOM, err = sbomContent.InFormats(context.BuildpackInfo.SBOMFormats...)
-			if err != nil {
-				return packit.BuildResult{}, err
-			}
-		} else {
-			logger.Process("Reusing cached layer %s", nodeModulesLayer.Path)
-			err := os.RemoveAll(filepath.Join(projectPath, "node_modules"))
+			run, sha, err := process.ShouldRun(projectPath, layer.Metadata, globalNpmrcPath)
 			if err != nil {
 				return packit.BuildResult{}, err
 			}
 
-			err = os.Symlink(filepath.Join(nodeModulesLayer.Path, "node_modules"), filepath.Join(projectPath, "node_modules"))
-			if err != nil {
-				return packit.BuildResult{}, err
+			if run {
+				logger.Process("Executing launch environment install process")
+
+				layer, err = layer.Reset()
+				if err != nil {
+					return packit.BuildResult{}, err
+				}
+
+				if build {
+					err := fs.Copy(filepath.Join(buildLayerPath, "node_modules"), filepath.Join(projectPath, "node_modules"))
+					if err != nil {
+						return packit.BuildResult{}, err
+					}
+					process = pruneProcess
+				}
+
+				duration, err := clock.Measure(func() error {
+					return process.Run(layer.Path, npmCacheLayer.Path, projectPath, globalNpmrcPath, true)
+				})
+				if err != nil {
+					return packit.BuildResult{}, err
+				}
+
+				if build {
+					err = fs.Move(filepath.Join(projectPath, "node_modules"), filepath.Join(layer.Path, "node_modules"))
+					if err != nil {
+						return packit.BuildResult{}, err
+					}
+
+					err = os.Symlink(filepath.Join(buildLayerPath, "node_modules"), filepath.Join(projectPath, "node_modules"))
+					if err != nil {
+						return packit.BuildResult{}, err
+					}
+
+				}
```

Comment:
Do we not want to do this in all cases?

Reply:
This is handled in all other install processes outside of `prune` and because it requires know two layer directories as opposed to just the one that information cannot be easily passed to the `prune` process.